### DISCUSSION TITLE
Add setReady method to ShulkerServerAPI

### DISCUSSION THREAD

**Discussion by Andrei965** [jeremylvln/Shulker#808] [Ideas]
I am currently running a Minestom server using ShulkerServerAgentMinestom. After setting up my game, I intend to reuse the server once the game ends. However, the ShulkerServerAPI currently exposes only the methods setAllocated, setReserved, and askShutdown, and does not include a setReady method. To address this, I propose adding the setReady method, which would enable the server to be reused once the game ends.

**Comment by jeremylvln** [comment]
Hi @Andrei965, that's a fair suggestion. Will this be enough for you to re-use the game server or do you need the agent to do something else too?

I'll try to add this ASAP

**Reply by Andrei965** [reply]
Thank you! I believe adding setReady will suffice for now. However, it would be helpful to expose additional methods to provide users with greater flexibility.

**Reply by Andrei965** [reply]
Hi! Would it be possible to add the [WatchGameServer](https://agones.dev/site/docs/guides/client-sdks/#watchgameserverfunctiongameserver) method? This would enable passing information to the game server during allocation.

**Reply by Andrei965** [reply]
Hi @jeremylvln! I've created PR #833, which introduces the watch_game_server method. Could you please review it? Additionally, would it be possible for you to add counters and lists to the [allocate_from_fleet](https://github.com/jeremylvln/Shulker/blob/d0dce981e26ec6bce71b4ac689984c281d3a16df/packages/shulker-operator/src/api/sdk_grpc.rs#L25) function? This would enable me to run [high-density game servers](https://agones.dev/site/docs/integration-patterns/high-density-gameservers/). For Minecraft game servers, running multiple games, each  in its own world, on each server is more efficient, and the changes would help with that.

**Reply by jeremylvln** [reply]
Hi @Andrei965 I never took the time to look into high-density game servers. But it looks really nice, actually. I don't see why I could not add support for them. I will read carefully the documentation to be sure I'm not missing anything

**Reply by Andrei965** [reply]
Hi @jeremylvln! What's the status on high-density game servers and #833? Also, thanks a lot for fixing the allocateFromFleet bug!
